src/librustc/session/config.rs

```diff
@@ -2288,10 +2304,25 @@ pub fn build_session_options_and_crate_config(
         )
     }
 
-    let extern_private = matches.opt_strs("extern-private");
+    let mut extern_private: BTreeMap<_, BTreeSet<_>> = BTreeMap::new();
+
+    for arg in matches.opt_strs("extern-private").into_iter() {
```

Comment:
If `ExternPrivates` are merged into `Externs` then perhaps these two loops could be deduplicated as well.